Use pod anti-affinity in the worker Job template spec

marquiz opened the pull request:
Use pod anti-affinity to explicitly indicate that pods should be
scheduled on separate nodes. And, remove the obscure and non-obvious
ports definition whose only purpose was to reach the same effect.

marquiz commented:
Rebased on latest master, adapted to the master/worker split

/remove-lifecycle stale

Ethyling commented on nfd-worker-job.yaml.template:
```diff
@@ -14,6 +14,16 @@ spec:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: kubernetes.io/hostname
+              labelSelector:
+              matchExpressions:
```
You need to add one more indent on the `matchExpressions` field

Ethyling replied:
@marquiz I don't know if you missed this one

marquiz replied:
> @marquiz I don't know if you missed this one

Yup, seems I did. Thanks for spotting this! Fixed, now

Ethyling commented:
@marquiz Sorry I missed that, but you also need to remove the comment in the `label-nodes.sh` script

marquiz commented:
> @marquiz Sorry I missed that, but you also need to remove the comment in the `label-nodes.sh` script

Np, thanks for spotting this, also! Now fixed. I just removed the comment, instead of replacing/updating it in order to avoid similar "out-of-sync" issues in the future

Ethyling commented:
/lgtm
